osquery/tables/system/linux/processes.cpp

```diff
@@ -56,6 +58,32 @@ inline std::string readProcCMDLine(const std::string& pid) {
   return content;
 }
 
+inline std::string readProcCgroup(const std::string& pid) {
+  auto attr = getProcAttr("cgroup", pid);
+
+  std::string content;
+  readFile(attr, content);
+  // Get only the first line
+  // with v1 cgroups we'll have separate lines for different cgroup types
+  if (auto pos = content.find_first_of('\n'); pos != std::string::npos) {
+    content.erase(pos);
+  }
+  boost::algorithm::trim_right_if(content,
+                                  [](const char& c) { return c == '\n'; });
+
+  // We should always get something like:
+  // 0::user.slice (for cgroup v2) or
+  // 2:cpu:user.slice (for cgroup v1)
+  // Note that a cgroup name may have colons
+  std::vector<std::string> fields;
+  boost::split(fields, content, [](const char& c) { return c == ':'; });
+  if (fields.size() < 3) {
+    return "";
+  }
+
+  return boost::join(std::vector(fields.begin() + 2, fields.end()), ":");
```

Comment:
Is this `boost::split` + `boost::join` returning a substr from the second `:` character until the end?

Reply:
Yes. I spent a while looking for a split function where I could tell it to split into exactly 3 but I couldn't find one